### ISSUE TITLE
When multiple repos are defined artifacts not resolved.

### ISSUE THREAD

**Issue by msavy** [snyk/snyk-maven-plugin#9]
For a reproducer of this, please clone: https://github.com/Apicurio/apicurio-studio.git
And check out: 03bb43a3600a2e1cc1b13838bf8baafd8b42689d
Do a `mvn clean install` if necessary.
Then attempt `snyk test` (which will fail as it can't resolve artifacts).

If you check the top level pom, you will notice that there are multiple repositories defined (both in the top-level repositories section and pluginRepositories). 

If you comment out the JBoss repos (https://gist.githubusercontent.com/msavy/111a7385cb424f5ab052e7d86a74946a/raw/476eaf074b00e8d589946a84c891124fb18dba5f/-) the build and scan work properly.

The issue seems to be that Snyk only uses the JBoss repos, which don't contain all the artifacts needed to successfully build the project. Normally a build falls back to alternative repositories if it can't locate them in the first.

**Comment by msavy** [comment]
I'm using the standard Snyk CLI for this, incidentally, which seems to call onto this plugin AFAIK.

**Comment by darscan** [comment]
Hi @msavy 

I've been unable to reproduce the issue. I did the following:
```
git clone git@github.com:Apicurio/apicurio-studio.git
cd apicurio-studio
git checkout 03bb43a3600a2e1cc1b13838bf8baafd8b42689d
mvn install -DskipTests
snyk test
```
And I got
```
✓ Tested 120 dependencies for known vulnerabilities, no vulnerable paths found.
```
Confusingly, this repo (`snyk-maven-plugin`) hosts an actual `Maven` plugin (not a CLI plugin).

The CLI uses https://github.com/snyk/snyk-mvn-plugin - which is a simple wrapper around `mvn dependency:tree -DoutputType=dot` - which uses Maven's dependency resolver (complete with repositories).

How does the issue you are seeing manifest? Is there some kind of error output?

**Comment by msavy** [comment]
I'm honestly slightly baffled!

It definitely wasn't working before (and a couple of us tried it) -  always getting the symptoms described in the original post until we applied the fix of removing the JBoss repos from our POMs.

I was previously able to reproduce the issue on another OSS project I work on apiman/apiman (before removing the repos there, also). FWIW, apiman doesn't seem to work in the [Snyk web application](https://snyk.io/org/apicurio/project/0277ff86-fda9-489b-a40b-67bd4b29e93f/history/e1eda42c-782a-4f3a-b56b-d848f4aa2c87). 

But, seemingly I can't repro it any more. The only thing I can think is that something changed  behind the scenes, or we screwed up something in a strange way :-).

Unfortunately I don't think I have any of the build logs hanging around, as I foolishly assumed it would continue being a 100% reproducible issue.

The only thing I _can_ reproduce now: if I remove the latest snapshot artifacts from ~/.m2, and don't manually do a maven build first, then it complains about missing snapshot artifacts. That's a different thing entirely, though.

I'll close the issue if I can't get a useful reproducer for you shortly.
